addons/udes_stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -983,3 +1004,196 @@ def remove_unfinished_work(self):
+                )
+
+            move_lines = move_lines.filtered(
+                lambda ml: ml.product_id == product and ml.location_id == location
+            )
+
+            msg = _("Unpickable product %s at location %s") % (product.name, location.name)
+
+            if lot_name:
+                lot_name = [lot_name] if isinstance(lot_name, str) else lot_name
+                move_lines = move_lines.filtered(lambda ml: ml.lot_id.name in lot_name)
+                msg += _(" with serial number %s") % lot_name
+
+            if package:
+                move_lines = move_lines.filtered(lambda ml: ml.package_id == package)
+                msg += _(" in package %s") % package.name
+            elif move_lines.package_id:
+                raise ValidationError(
+                    _("Unpickable product from a package but no package name provided.")
+                )
+
+            # at this point we should only have one move_line
+            quants = move_lines.get_quants()
+            allow_partial = True
+        elif package:
+            if not location:
+                location = package.location_id
+
+            move_lines = move_lines.filtered(
+                lambda ml: ml.package_id == package or ml.package_id.parent_id == package
+            )
+            quants = package._get_contained_quants()
+            msg = _("Unpickable package %s at location %s") % (package.name, location.name)
+        else:
+            raise ValidationError(
+                _("Missing required information for unpickable item: product or package.")
+            )
+
+        # Filter out the move lines already picked in case under received
+        move_lines = move_lines.filtered(lambda ml: ml.qty_done == 0)
+
+        if not move_lines:
+            raise ValidationError(
+                _("Cannot find move lines todo for unpickable item in this batch.")
+            )
+
+        pickings = move_lines.picking_id
+        original_picking_ids = {}
+        if raise_stock_investigation:
+            to_investigate = Picking.browse()
+        for picking in pickings:
+            picking.message_post(body=msg)
+
+            if picking.batch_id != self:
+                raise ValidationError(_("Move line is not part of the batch."))
+
+            if picking.state in ["cancel", "done"]:
+                raise ValidationError(
+                    _(
+                        "Cannot mark a move line as unpickable "
+                        "when it is part of a completed Picking."
+                    )
+                )
+
+            original_id = None
+
+            if len(picking.move_line_ids - move_lines):
+                # Create a backorder for the affected move lines if there are other move
+                # lines not relevant to unpickable move lines
+                original_id = picking.id
+                picking = picking._backorder_move_lines(picking.move_line_ids - move_lines)
```

Comment:
backorder = picking._...

Reply:
It might seem logical to do that but the line here is specifically reassigning the picking variable such that the following code acts on the original picking or backorder picking depending on the case. I will try to see if I can make this a little clearer but I cannot simply rename to backorder.